File: sqrt_cpp/sqrt.cpp

#include <iostream>
#include <math.h>
using namespace std;

float myFabs(float x){
    return x < 0 ? -x : x;
}

float mySqrt(float val){
    float x = val/10;
    float dx;
    double diff;
    double min_tol = .00001;
    bool flag = false;

    if(val == 0){
        x = 0;
    }else{
        for(int i = 1; i <20; i++){
            if(!flag){
                dx = (val - pow(x, 2)) / (2 * x);
                x += dx;
                diff = val - pow(x, 2);
                flag = (myFabs(diff) <= min_tol);
            }
        }
    }
    return x;

}


int main(){
    float sqrt = mySqrt(123.123);

    cout << "sqrt(123.123) = " << sqrt << "\n";
}

File: sqrt/sqrt_rs/sqrt.rs



fn main(){
    println!("{}", my_sqrt(123.123));

}

fn my_fabs(x : f64) -> f64{
    if x < 0.0{
        -x
    }else{
        x
    }
}

fn my_sqrt(val:f64) -> f64{
    let mut x = val/10.0;
    let min_tol = 0.00001;
    let mut flag = false;

    if val == 0.0{
        0.0
    }else{
        for _i in 1..20{
            if !flag{
                let  dx = (val - x.powi(2)) / (2.0 * x);
                x += dx;
                let diff = val - x.powi(2);
                flag = my_fabs(diff) <= min_tol;

            }
        }
        x
    }
    
}